plugins/module_utils/network/sonic/config/ospf_area/ospf_area.py

```diff
@@ -0,0 +1,1147 @@
+        return formatted_area_policy
+
+    def format_ranges_to_rest(self, want):
+        '''format the ranges an adrea advertises into REST body format. Takes a list of ranges'''
+        formatted_ranges = []
+        for range_settings in want:
+            formatted_range_config = {}
+            if "advertise" in range_settings:
+                formatted_range_config["advertise"] = range_settings["advertise"]
+            if "cost" in range_settings:
+                formatted_range_config["metric"] = range_settings["cost"]
+            if "substitute" in range_settings:
+                # note it is mispelled in REST
+                formatted_range_config["substitue-prefix"] = range_settings["substitute"]
+            # can add range even without other settings
+            formatted_range_config["address-prefix"] = range_settings["prefix"]
+            formatted_ranges.append({"address-prefix": range_settings["prefix"], "config": formatted_range_config})
+        return formatted_ranges
+
+    def build_areas_delete_requests(self, commands, have, delete_everything=False):
+        '''takes in a list of areas and builds all required 'delete' requests for the specified areas
+        :param commands: list of areas to make delete requests for. assumed to be a subset of have
+        :param have: the current config in argspec format, at the top level of definition
+        :param delete_everything: whether to delete config for all area'''
+        requests = []
+        area_h_keys = {(area["area_id"], area["vrf_name"]): area for area in have}
+        for area_c in commands:
+            # want to match so can find out if commands specifies everything within existing area so can de a delete all of area
+            matched_have = area_h_keys.get((area_c["area_id"], area_c["vrf_name"]), None)
+            if not matched_have:
+                # would mean nothing to delete, just ignore. should not hit since commands should be a subset of have
+                continue
+            area_all_gone, area_delete_requests = self.build_area_delete_requests(area_c, matched_have, delete_everything)
+            requests.extend(area_delete_requests)
+        return requests
+
+    def build_area_delete_requests(self, commands, have, delete_everything=False):
+        '''builds the requests to delete configuration under the areas section of config for a single area. takes a pair of single area commands and have.
+        returns tuple of whether everything in area was deleted and requests to cause changes'''
+        requests = []
+        # can't go directly to deleting area, need to check for ranges, network, virtual links -
+        # most of nested and complex settings - and delete those first (and separately from area)
+        if len(commands) == 2 or delete_everything:
+            # allow specifying area id to mean clear it
+            delete_everything = True
+            commands = have
+        area_deleted = False
```

Comment:
Added comment into code itself
```
# while clearing subsections for area, area itself may get removed in certain cases, for example clearing virtual links when it is the only
# config inside area.
# This variable tracks if above has happened. It updates regardless of whether or not module clears all settings for area, but
# is only used in case of clearing all settings and area. This prevents module from making a second unnecessary and 
# error-causing request in cases where area is removed early.
```

Reply:
The code revisions to address this comment look good.

Thank you for enhancing the descriptions to help code maintainers understand this usage more clearly.

Resolved.